x-pack/plugins/infra/public/components/asset_details/tabs/metadata/add_pin_to_row.tsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { Dispatch } from 'react';
+import { i18n } from '@kbn/i18n';
+import { EuiToolTip, EuiButtonIcon } from '@elastic/eui';
+import type { Field } from './utils';
+
+interface AddMetadataPinToRowProps {
+  fieldName: Field['name'];
+  pinnedItems: Array<Field['name']>;
+  setPinnedItems: Dispatch<React.SetStateAction<Array<Field['name']> | undefined>>;
+}
+
+const PIN_FIELD = i18n.translate('xpack.infra.metadataEmbeddable.pinField', {
+  defaultMessage: 'Pin Field',
+});
+
+export const AddMetadataPinToRow = ({
+  fieldName,
+  pinnedItems,
+  setPinnedItems,
```

Comment:
Since this is an event handler, I'd name it `onPinned` or something along those lines. Wdyt?

Reply:
Hmm this is basically setting the the pinned items to the locale storage [here](https://github.com/elastic/kibana/pull/161074/files#diff-ea25857eb8e5e15cea05d3b7eadd330dc7cc1815fbc185be28060d184cad6c7dR74) but in this component I can rename it 👍

Reply:
Renamed ✅